pkg/webhook/policy.go

```diff
@@ -286,7 +289,14 @@ func (h *validationHandler) reviewRequest(ctx context.Context, req admission.Req
 	if req.AdmissionRequest.Namespace != "" {
 		ns := &corev1.Namespace{}
 		if err := h.client.Get(ctx, types.NamespacedName{Name: req.AdmissionRequest.Namespace}, ns); err != nil {
-			return nil, err
+			if !k8serrors.IsNotFound(err) {
+				return nil, err
+			}
+			// bypass cached client and ask api-server directly
+			err = h.reader.Get(ctx, types.NamespacedName{Name: req.AdmissionRequest.Namespace}, ns)
```

Comment:
To bypass cache and to reduce impact on the api server, is it better to use `client.New` or `GetAPIReader`?

Reply:
GetAPIReader avoids creating a new RESTMapper, whereas client.New() re-runs the discovery client.